### DISCUSSION TITLE
What's the intended way to create a conditional schema?

### DISCUSSION THREAD

**Discussion by mwskwong** [open-circle/valibot#771] [Q&A] [answered]
Consider the following schema:

```ts
const EnvSchema = object({
  CRON_SECRET: process.env.VERCEL === '1' ? string() : optional(string())
})
```

Basically, the idea is if the application is deployed to Vercel, ensure that `CRON_SECRET` is set. Otherwise, it is optional.

The above seems to function as intended, but then I noticed `lazy()`, which seems to be aimed at solving problems like this. What is the intended way to implement a conditional schema? Should I use `lazy()` in this case?

**Comment by fabian-hiller** [comment, marked as answer]
Your approach is perfectly fine. I would write it the same way. You only need `lazy` for recursive schemas, or if you create a schema depending on the input or an external value that may change over time. But `process.env.VERCEL` is most likely a static value.
